src/server/deploy-service/controllers/deploy/deploy.js

```diff
@@ -36,30 +36,35 @@ const deployController = {
         }
       )
 
-      if (response?.ok) {
-        // Remove step data session
-        request.yar.clear(multiStepFormId)
+      if (!response?.ok) {
```

Comment:
A few pointers around this code (if its needed)

- 1) This is saying. If `response.ok` is falsey -  `false`, `0`, '""`, `null` etc - https://developer.mozilla.org/en-US/docs/Glossary/Falsy

```
const response = { ok: false }

if (!response?.ok) {
 // true
}
```
also
```
const response = {}

if (!response?.ok) {
 // true
}
```
and
```
const response = { ok: 0 }

if (!response?.ok) {
 // true
}
```
etc...

- 2) It then goes on to fall through. Not sure If I like this TBH as its a different pattern to all other `POST` controllers in the PFE. So would be quite confusing
- 3) Is this actually working as intended?

## Suggested updates

- If this work is needed? (need to understand whats been written here so will catch up on Monday with you)
- I would wrap the positive flow as it was previously in: 
```
if (response?.ok) {
 ...
```
- If you have something that should happen on a negative response go for:
```
if (response?.ok === false) {
  ...
```

Reply:
It would have to be 
```
if (response === undefined || response.ok === false) {
```
which is wordier than
```
if(!response?.ok){
```
as previously if response failed, it was not catching that.

Reply:
Will explain why in a huddle